### ISSUE TITLE
Was it right way to create new DataType for qualifiers?

### ISSUE THREAD

**Issue by ptomin** [uxmal/reko#701]
I think, adding `QualifiedType` is not right way. Adding qualifier to type does not makes it other type
e.g. `const char` should be instance of `PrimitiveType` rather than `QualifiedType`.
Using `QualifiedType` cause checks like `dt is PrimitiveType`,  `dt is Pointer`,  `dt is Structure` be incorrect.
It could be fixed off course, but I think better way is adding new `qualifiers` field to `DataType`

**Comment by uxmal** [comment]
You're right; it's probably better to add a `Qualifiers` field to `DataType`, allowing for `Constant` and `Volatile`, and the combination of the two. It means adding a field to `DataType` as you say. I can do this tonight and see what falls out.

**Comment by uxmal** [comment]
Pavel, I've pushed the branch `cv-qualifiers`. Could you review this?
